sklearn/cluster/_agglomerative.py

```diff
@@ -453,8 +453,13 @@ def linkage_tree(X, connectivity=None, n_clusters=None, linkage='complete',
         if affinity == 'precomputed':
             # for the linkage function of hierarchy to work on precomputed
             # data, provide as first argument an ndarray of the shape returned
-            # by pdist: it is a flat array containing the upper triangular of
-            # the distance matrix.
+            # by sklearn.metrics.pairwise_distances.
+            if len(X.shape) != 2 or X.shape[0] != X.shape[1]:
+                raise ValueError(
+                    'Matrix should be square, '
+                    'as returned by pairwise_distances. '
+                    'Found dimensionality %s' % str(X.shape)
```

Comment:
I would not say as return by pairwise_distances, because pairwise distances can return non square matrix (if called on X and Y with different shapes). I would just say
```python
'Matrix should be squared. Found matrix of shape {}'.format(X.shape)
```

Reply:
Done. Just added 'Distance matrix' instead of 'Matrix' to be more specific.

Reply:
Please use the `.format` syntax.

Reply:
Done.